SensusService/Probes/User/ScriptRunner.cs

```diff
@@ -516,7 +541,7 @@ private void StartRandomTriggerCallbacksAsync()
                         _randomTriggerCallbackId = SensusServiceHelper.Get().ScheduleOneTimeCallback((callbackId, cancellationToken) =>
                             {
                                 // if the probe is still running and the runner is enabled, run a copy of the script so that we can retain a pristine version of the original
-                                if (_probe.Running && _enabled && _randomTriggerWindows.Any(window => DateTime.Now.Hour >= window.Item1 && DateTime.Now.Hour <= window.Item2))  // be sure to use DateTime.Now and not the local now variable, which will be in the past.
+                                if (_probe.Running && _enabled && _randomTriggerWindows.Any(window => ((DateTime.Now >= window.Item1 && DateTime.Now <= window.Item2) || (window.Item1 == window.Item2 && DateTime.Now.Hour == window.Item1.Hour && DateTime.Now.Minute == window.Item1.Minute))))  // be sure to use DateTime.Now and not the local now variable, which will be in the past.
```

Comment:
Similar problem here, too...I think. Your comparisons of `DateTime.Now >= window.Item1` and `DateTime.Now >= window.Item1` will become permanently false at some point in the future.